include/uart.h

```diff
@@ -257,6 +318,58 @@ static inline unsigned char _impl_uart_poll_out(struct device *dev,
 	return api->poll_out(dev, out_char);
 }
 
+/**
+ * @brief Set UART configuration.
+ *
+ * Sets UART configuration using data from *cfg.
+ *
+ * @param dev UART device structure.
+ * @param cfg UART configuration structure.
+ *
+ *
+ * @retval 0 If successful.
+ * @retval -ENOTSUP If configuration is not supported by device.
```

Comment:
Please specify that "<0" means error, and only then describe particular ones, giving clear hint that there can be more. E.g. something like "<0 error, including: -ENOTSUP ...". Can also look thru other docstrings with similar formulations for ideas.

Reply:
I wanted to use ENOSYS if function is not implemented, but got some problems, see here #11207

Reply:
> I wanted to use ENOSYS if function is not implemented, but got some problems, see here

Please use whatever other APIs use in this case, and that's ENOTSUP. If you'd like to change it, it should be done consistently across API, and that's a separate task (and quite a task, with a usual RFC phase, etc.)

Reply:
Changed docstrings `@retval 0 If successful, negative errno code otherwise.`